src/endpoints/unit/sell-products/selling-controller.ts

```diff
@@ -0,0 +1,114 @@
+import {Body, Controller, Post} from '@nestjs/common';
+import {SaleService} from 'src/endpoints/sale/sale.service';
+import {Sale, SalePaymentMethod, SaleProduct, SaleRequest} from 'src/models';
+import {ClsService} from 'nestjs-cls';
+import {v4 as uuidv4} from 'uuid';
+import {SaleProductService} from 'src/endpoints/sale-product/sale-product.service';
+import {ProductService} from 'src/endpoints/product/product.service';
+import {SalePaymentMethodService} from 'src/endpoints/sale-payment-method/sale-payment-method.service';
+import {error} from 'console';
+import {PaymentMethodService} from 'src/endpoints/payment-method/payment-method.service';
+
+@Controller('api/selling')
+export class SellingController {
+  constructor(
+    private readonly saleService: SaleService,
+    private readonly cls: ClsService,
+    private readonly saleProductService: SaleProductService,
+    private readonly productService: ProductService,
+    private readonly salePaymentMethodService: SalePaymentMethodService,
+    private readonly paymentMethodService: PaymentMethodService,
+  ) {}
+
+  @Post('sellProducts')
+  async sellProducts(@Body() saleRequest: SaleRequest) {
+    const prices = saleRequest.products.map(
+      x => x.sellingPrice * x.productCount,
+    );
+    const total = prices.reduce(
+      (accumulator, currentValue) => accumulator + currentValue,
+    );
+    const products = await this.productService.getAll();
```

Comment:
Create a new method in the ProductService called `getByIds(ids:string[])` and load only ids available in the request. Read about [In](https://typeorm.io/find-options#advanced-options)